components/BookCard.tsx

```diff
@@ -0,0 +1,69 @@
+import { Box, Heading, Image, Text } from "@chakra-ui/react";
+import { useBookQuery } from "@lib/hooks/book";
+import { Book } from "@prisma/client";
+import { useRouter } from "next/router";
+
+export default function BookCard({ name, imageUrl, isbn }: Book) {
+  const { isLoading, data: populatedBook } = useBookQuery(isbn);
+  const router = useRouter();
+
+  const authors = (): string => {
+    if (isLoading) {
+      return "Loading Authors..."
```

Comment:
Checkout this component https://chakra-ui.com/docs/feedback/skeleton. Might make for a nicer experience.

Reply:
Had a lot fun with these 👍